packages/xml-views-validation/src/validators/document/non-unique-id.ts

```diff
@@ -80,11 +93,19 @@ class IdsCollectorVisitor implements XMLAstVisitor {
       isPossibleCustomClass(attrib.parent)
     ) {
       if (this.idsToXMLElements[attrib.value] === undefined) {
-        // @ts-expect-error - TSC does not understand: `attrib.syntax.value !== undefined` is a type guard
-        this.idsToXMLElements[attrib.value] = [attrib];
+        this.idsToXMLElements[attrib.value] = [
+          {
+            // @ts-expect-error - TSC does not understand: `attrib.syntax.value !== undefined` is a type guard
```

Comment:
Did you try assigning `this.idsToXMLElements[attrib.value]` to a local variable before the check? That usually works.

Reply:
seems obsolete after refactoring.